fennel/testing/executor.py

```diff
@@ -482,56 +482,94 @@ class WindowStruct:
             def __init__(self, event_time: datetime):
                 self.begin_time = event_time
                 self.end_time = event_time + timedelta(microseconds=1)
-                self.count = 1
 
             def add_event(self, event_time: datetime):
-                self.count += 1
                 self.end_time = event_time + timedelta(microseconds=1)
 
             def to_dict(self) -> dict:
                 return {
                     "begin": self.begin_time,
                     "end": self.end_time,
-                    "count": self.count,
                 }
 
         def generate_sessions_for_df(
-            df: pd.DataFrame, timestamp_col: str, gap: int, field: str
+            df: pd.DataFrame,
+            timestamp_col: str,
+            gap: int,
+            field: str,
+            summary: Optional[Summary],
         ):
             # Sort the DataFrame by timestamp_col
             df = df.sort_values(by=timestamp_col)
-
             # Initialize lists to store session information
             window_list: List[dict] = []
             timestamp_list: List[datetime] = []
             current_window: Optional[WindowStruct] = None
+            rows: List = []
+            summary_value = []
 
             # Iterate through the rows of the DataFrame
-            for index, row in df.iterrows():
+            for _, row in df.iterrows():
                 # Check if it's the first event
                 if current_window is None:
                     current_window = WindowStruct(row[timestamp_col])
+                    rows.append(row)
                 else:
                     # Check if the event is within the time threshold of the previous one
                     if (
                         row[timestamp_col]
                         - (current_window.end_time - timedelta(microseconds=1))
                     ).total_seconds() <= gap:
                         current_window.add_event(row[timestamp_col])
+                        rows.append(row)
                     else:
                         # Save window information and reset for a new window
                         window_list.append(current_window.to_dict())
                         timestamp_list.append(current_window.end_time)
+                        if summary is not None:
+                            value = summary.summarize_func(pd.DataFrame(rows))
+                            summary_value.append(value)
                         current_window = WindowStruct(row[timestamp_col])
+                        rows = [row]
 
             # Save the last window information
+            # Save the window information
             window_list.append(current_window.to_dict())  # type: ignore
             timestamp_list.append(current_window.end_time)  # type: ignore
+            if summary is not None:
+                value = summary.summarize_func(pd.DataFrame(rows))
```

Comment:
You are collecting all the rows and creating the same dataframe ? What is the point of this ?

Reply:
I collect all the rows of one window to summarize